puffin/src/claims.rs

```diff
@@ -11,35 +10,38 @@ use crate::agent::AgentName;
 use crate::algebra::dynamic_function::TypeShape;
 use crate::protocol::{EvaluatedTerm, ProtocolTypes};
 
-pub trait Claim<PT: ProtocolTypes>: EvaluatedTerm<PT> + Debug {
+pub trait Claim: EvaluatedTerm<Self::PT> + Debug {
+    type PT: ProtocolTypes;
+
     fn agent_name(&self) -> AgentName;
-    fn id(&self) -> TypeShape<PT>;
-    fn inner(&self) -> Box<dyn EvaluatedTerm<PT>>;
+    fn id(&self) -> TypeShape<Self::PT>;
+    fn inner(&self) -> Box<dyn EvaluatedTerm<Self::PT>>;
 }
 
-pub trait SecurityViolationPolicy<PT: ProtocolTypes, C: Claim<PT>> {
-    fn check_violation(claims: &[C]) -> Option<&'static str>;
+pub trait SecurityViolationPolicy {
+    type C: Claim;
```

Comment:
Question: I guess the goal is to lighten the type annotations and ease potential future git conflicts?

Reply:
Yes, also objects implementing this trait needs to operate on a single Claim type.